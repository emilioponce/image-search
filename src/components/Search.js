import React, { Component } from 'react'

const Search = () => {
  return (
    <div>
      Key word: <input type="text" />
    </div>
  )
}

export default Search
