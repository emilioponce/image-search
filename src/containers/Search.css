.input {
  font-size: 20px;
  height: 30px;
  width: 400px;
  color: #3a4458;
}
