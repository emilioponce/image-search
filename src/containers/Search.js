import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchImages } from '../actions'

import './Search.css'

class Search extends Component {
  constructor() {
    super()
    this.state = {
      keyword: ''
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(event) {
    this.setState({ keyword: event.target.value })
  }

  handleSubmit(event) {
    event.preventDefault()
    const { keyword } = this.state
    if (keyword !== '') {
      this.props.fetchImages(keyword)
      this.setState({ keyword: '' })
    }
  }

  render() {
    const { keyword } = this.state
    return (
      <form onSubmit={this.handleSubmit}>
        <div className="wrapperSearch">
          <input
            type="text"
            value={keyword}
            onChange={this.handleChange}
            placeholder="Search a term ..."
            className="input"
          />
          <button type="submit" className="button ">
            SEARCH
          </button>
        </div>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchImages: keyword => dispatch(fetchImages(keyword))
  }
}

export default connect(
  null,
  mapDispatchToProps
)(Search)
