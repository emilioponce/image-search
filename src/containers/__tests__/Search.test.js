import React from 'react'
import ReactDOM from 'react-dom'

import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// Strategy: testing the component 'Search' ALONE, without being connected to redux
// more info: https://hackernoon.com/unit-testing-redux-connected-components-692fa3c4441c
import { Search } from '../Search'

configure({ adapter: new Adapter() })

const mockModifyKeywordFn = jest.fn()
const mockFetchImagesFn = jest.fn()

const component = (
  <Search
    keyword={'cats'}
    modifyKeyword={mockModifyKeywordFn}
    fetchImages={mockFetchImagesFn}
  />
)

describe('Search component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(component)
  })

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matches', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call the fetch Images function once', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} })
    expect(mockFetchImagesFn.mock.calls.length).toBe(1)
  })

  it('Should let the user introduce text in the input field', () => {
    const changeEvent = {
      target: { value: 'cats' }
    }
    wrapper.find('input').simulate('change', changeEvent)
    expect(wrapper.find('input').props().value).toEqual('cats')
    expect(mockModifyKeywordFn.mock.calls.length).toBe(1)
  })
})
