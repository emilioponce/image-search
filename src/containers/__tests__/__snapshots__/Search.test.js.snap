// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Snapshot matches 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="wrapperSearch"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search a term ..."
      type="text"
      value="cats"
    />
    <button
      className="button "
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
